pydantic_graph/pydantic_graph/_utils.py

```diff
@@ -84,7 +63,11 @@ def get_parent_namespace(frame: types.FrameType | None) -> dict[str, Any] | None
     """
     if frame is not None:
         if back := frame.f_back:
-            if back.f_code.co_filename.endswith('/typing.py'):
+            if frame.f_globals.get('__name__') == 'typing':
+                # If the class calling this function is generic, explicitly parameterizing the class
+                # results in a `typing._GenericAlias` instance, which proxies instantiation calls to the
+                # "real" class and thus adding an extra frame to the call. To avoid pulling anything
+                # from the `typing` module, use the correct frame (the one before):
```

Comment:
I aligned this with how we do it [in Pydantic](https://github.com/pydantic/pydantic/blob/98348d0ce8e26c42ac9d32278ef97f6545446dd9/pydantic/type_adapter.py#L237-L242)